src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -250,5 +250,69 @@ void IEnumerator.Reset()
 
             void IDisposable.Dispose() { }
         }
+
+        public bool TryGetValue(string key, out Group value)
+        {
+            if (ContainsKey(key))
+            {
+                value = this[key];
+                return true;
+            }
+            value = null;
+            return false;
+        }
+
+        public bool ContainsKey(string key)
+        {
+            return _match._regex.GroupNumberFromName(key) >= 0;
+        }
+
+        public IEnumerable<string> Keys
+        {
+            get
+            {
+                return _match._regex.GetGroupNames();
+            }
+        }
+
+        public IEnumerable<Group> Values
+        {
+            get
+            {
+                return this._groups;
+            }
+        }
+
+        private sealed class EnumeratorKeyValue : IEnumerator<KeyValuePair<string, Group>>
+        {
+            private Enumerator _collectionEnumerator;
+
+            public EnumeratorKeyValue(GroupCollection collection)
+            {
+                _collectionEnumerator = new Enumerator(collection);
```

Comment:
This means that enumerating this as an IReadOnlyDictionary will now incur two allocations instead of one.  There's minimal logic in the existing enumerator; I think it'd be better to use copy that logic here, updated accordingly, to avoid that second enumerator allocation.